### PR TITLE
Adds additional e2e tests and fixes existing ones

### PR DESCRIPTION
**Pull request by dylankelly**
**JIRA issue:** https://digital-engagement.atlassian.net/browse/SDPA-

### Changed

1.

### CODE REVIEW

File: examples/basic-examples/package.json [hunk 1]
```diff
@@ -42,9 +42,6 @@
     "babel-preset-vue": "^2.0.2",
     "core-js": "^2.6.5",
     "cross-env": "^5.2.0",
-    "cypress": "^3.1.5",
-    "cypress-axe": "^0.4.0",
-    "cypress-cucumber-preprocessor": "^1.11.0",
```
Comment (tim-yao) [inline examples/basic-examples/package.json]: @dylankelly I think this need to be in our update script so create ripple app can update the project package.json file automatically.

File: examples/vic-gov-au/test/e2e/plugins/index.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
 
+let momentNow
+
 module.exports = (on, config) => {
   on('file:preprocessor', cucumber())
 
-  tideAdmin(on, config)
-
   config.env = {
+    SITE_ID: '4',
```
Comment (tim-yao) [inline examples/vic-gov-au/test/e2e/plugins/index.js]: Do we need this in each project? If so we need to make it configurable, as most of the project site id is not `4`.

File: packages/ripple-nuxt-tide/lib/components/AppForm.vue [hunk 3]
```diff
@@ -1,5 +1,5 @@
 <template>
-  <rpl-form v-if="formData" :formData="formData" :submitHandler="submitForm" :hideAfterSuccess="formData.settings.shouldHideFormAfterSuccess" :title="title" ></rpl-form>
+  <rpl-form v-if="formData" novalidate :name="`form-${formData.tideId}`" :formData="formData" :submitHandler="submitForm" :hideAfterSuccess="formData.settings.shouldHideFormAfterSuccess" :title="title" ></rpl-form>
```
Comment (tim-yao) [inline packages/ripple-nuxt-tide/lib/components/AppForm.vue]: Hey @dylankelly Is there a particular reason to use `novalidate` on webform?

File: .circleci/config.yml [hunk 4]
```diff
@@ -3,7 +3,7 @@ version: 2.0
 jobs:
   build:
     docker:
-      - image: cypress/base:10
+      - image: cypress/base:12.0.0
         environment:
```
[the remark this answers is not in the document]
Comment (dylankelly) [reply .circleci/config.yml]: This should be added to sdp-ripple-tasks

File: examples/vic-gov-au/package.json [hunk 5]
```diff
@@ -9,8 +9,8 @@
     "build:default": "if [ \"$NUXT_HOT_RELOADING\" != \"1\" ] ; then nuxt build; fi",
     "build:win32": "IF \"%NUXT_HOT_RELOADING%\" NEQ \"1\" ( nuxt build )",
     "cy:open": "cypress open",
-    "cy:run": "cypress run -e TAGS='not @skip or @smoke'",
-    "cy:run-record": "cypress run -e TAGS='not @skip or @smoke' --record  --parallel --group $CIRCLE_JOB",
+    "cy:run": "cypress run -b chrome -e TAGS='not @skip or @smoke'",
+    "cy:run-record": "cypress run -b chrome -e TAGS='not @skip or @smoke' --record  --parallel --group $CIRCLE_JOB",
```
[the remark this answers is not in the document]
Comment (dylankelly) [reply examples/vic-gov-au/package.json]: Added to create-ripple-app

File: examples/vic-gov-au/cypress.json [hunk 6]
```diff
@@ -6,7 +6,12 @@
...
-  "ignoreTestFiles": "*.js",
+  "ignoreTestFiles": [
+    "*.js",
+    "*.md"
+  ],
+  "snapshotFileName": "test/e2e/snapshots.js",
+  "projectId": "wtymar",
   "chromeWebSecurity": false,
-  "projectId": "wtymar"
+  "$schema": "https://on.cypress.io/cypress.schema.json"
```
[the remark this answers is not in the document]
Comment (dylankelly) [reply examples/vic-gov-au/cypress.json]: added to create-ripple-app

File: examples/vic-gov-au/test/e2e/plugins/index.js (hunk 2 above)
Comment (dylankelly) [reply examples/vic-gov-au/test/e2e/plugins/index.js]: added automatically with create-ripple-app

File: packages/ripple-nuxt-tide/lib/components/AppForm.vue (hunk 3 above)
Comment (dylankelly) [reply packages/ripple-nuxt-tide/lib/components/AppForm.vue]: yes, the html validation is inconsistent with VFG validation, flaky tests identified potentially buggy code.

File: packages/create-ripple-app/generators/updates/22.0.0/saofile.js [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+      'collections': true,
+      'paths': true,
+      'shorthands': true
+    }))
+    // Load moment 'en-au' locale only for performance.
+    // https://github.com/jmblog/how-to-optimize-momentjs-with-webpack
+    // You need to change it if your site is not in Australia.
+    config.plugins.push(new webpack.ContextReplacementPlugin(/moment[/\\\\]locale$/, /en-au/))
+  }
+}`
```
Comment (tim-yao) [inline packages/create-ripple-app/generators/updates/22.0.0/saofile.js]: Hey @dylankelly this is copied from 21 update script and need to be removed here.

File: packages/create-ripple-app/generators/updates/22.0.0/saofile.js [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+            }
+          )
+        }
+      })
+    }
+
+    return actions
+  },
+  async completed () {
+    console.log('Update to 21.0.0 complete!')
```
Comment (tim-yao) [inline packages/create-ripple-app/generators/updates/22.0.0/saofile.js]:
```suggestion
    console.log('Update to 22.0.0 complete!')
```

File: packages/create-ripple-app/generators/updates/22.0.0/saofile.js [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+          templateDir: `${templateDir}/_tests/_common`
+        }
+      )
+    }
+
+    if (results.smoke) {
+      actions.push({
+        type: 'add',
+        files: ['**'],
+        templateDir: `${templateDir}/_tests/_smoke`
```
Comment (tim-yao) [inline packages/create-ripple-app/generators/updates/22.0.0/saofile.js]: The smoke "tide-is-up" currently fails in a lot projects as there is no site id `4`. https://github.com/dpc-sdp/ripple/blob/feature/update-e2e-tests/packages/create-ripple-app/template/_tests/_smoke/test/e2e/integration/smoke/tide-is-up.feature#L7
Any idea to fix it?

File: packages/create-ripple-app/template/_tests/_common/test/e2e/integration/smoke/tide-is-up.feature [hunk 10]
```diff
@@ -4,10 +4,10 @@ Feature: Test Tide backend is up
 
   @smoke
   Scenario: Route endpoint returns a correct response
-    Given a request is made to "/api/v1/route?site=<%= siteid %>&path=/"
+    Given a request is made to "/api/v1/route?site=4&path=/"
```
Comment (tim-yao) [inline packages/create-ripple-app/template/_tests/_common/test/e2e/integration/smoke/tide-is-up.feature]: Id `4` will not work for those site id is not same. And some independent site has no site id 4. Any solution to make it dynamic?

File: packages/ripple-test-tools/package-lock.json [hunk 11]
```diff
@@ -0,0 +1,5754 @@
+{
```
Comment (tim-yao) [inline packages/ripple-test-tools/package-lock.json]: This shouldn't be added to project as yarn workplace should manage it in root yarn.lock file I think.

**Comment by dylankelly** [comment]
@tim-yao have updated again

File: packages/ripple-test-tools/package-lock.json (hunk 11 above)
Comment (dylankelly) [reply packages/ripple-test-tools/package-lock.json]: removed

File: packages/create-ripple-app/template/_tests/_common/test/e2e/integration/smoke/tide-is-up.feature (hunk 10 above)
Comment (dylankelly) [reply packages/create-ripple-app/template/_tests/_common/test/e2e/integration/smoke/tide-is-up.feature]: adding <%= siteid %> will copy in the siteid per project on update - just make sure its set when you run the update.

File: packages/create-ripple-app/generators/updates/22.0.0/saofile.js (hunk 7 above)
Comment (dylankelly) [reply packages/create-ripple-app/generators/updates/22.0.0/saofile.js]: removed

**Comment by tim-yao** [comment]
Hey @dylankelly This is a milestone of e2e tests. Just need CI passed. Somehow it's timeout at the moment.

**Pull request merged**